### DISCUSSION TITLE
Is there a JsonSchema for the JSON/Yaml rulesets?

### DISCUSSION THREAD

**Discussion by ouvreboite** [stoplightio/spectral#2666] [Ideas]
I’m looking to improve my (YAML) ruleset writing experience, and auto-completion would be nice.

Applying a JsonSchema to YAML file is easy in VSCode and provide some sort of autocompletion.

Does such a schema already exist ?

I started experimenting with one, trying to autogenerate one from the spectral/core interfaces and type, but the result is sub-optimal. I think hand-writing a schema could result in something decent but would require more work and would need to be kept in sync when/if the ruleset syntax change.

Such a schema could be registered in [schema store.org](https://www.schemastore.org/) to be automatically applied on `.spectral.json` and `.spectral.yml` files on supported IDEs

Would this be of interest?

**Comment by uuf6429** [comment]
If I'm not mistaken, what you mean can be found here:
https://github.com/stoplightio/spectral/tree/develop/packages/core/src/ruleset/meta

[It](https://raw.githubusercontent.com/stoplightio/spectral/refs/heads/develop/packages/core/src/ruleset/meta/ruleset.schema.json) seems to validate against the [example here](https://docs.stoplight.io/docs/spectral/01baf06bdd05a-create-a-ruleset#write-your-first-rule). The only problem is that it is referenced (and [references](https://github.com/stoplightio/spectral/blob/develop/packages/core/src/ruleset/meta/ruleset.schema.json#L18)) as a local file (presumably for when spectral is installed from npm). That means adding it to an arbitrary yaml/json file via `$schema` will not make it work in full.

On the other hand, building a full schema out of those files and submitting it schema store shouldn't be too hard a task..

PS: I'd also find this useful. 😊
PSS: ironically, schemastore are also use spectral: https://github.com/SchemaStore/schemastore/tree/master/config
